Main.java

```diff
@@ -3,74 +3,121 @@
-                    fileData = fileData.replaceAll(REMOVE_SPECIAL_CHAR, SINGLE_SPACE);
-                    StringTokenizer splitIntoToken = new StringTokenizer(fileData);
-                    while (splitIntoToken.hasMoreTokens()) {
-                        if (searchKeyword.equalsIgnoreCase(splitIntoToken.nextToken())) {
-                            wordCount++;
-                        }
+            while ((fileData = bufferedReader.readLine()) != null) {
+                fileData = fileData.replaceAll(REMOVE_SPECIAL_CHAR, SINGLE_SPACE);
+                StringTokenizer splitIntoToken = new StringTokenizer(fileData);
+                while (splitIntoToken.hasMoreTokens()) {
+                    if (searchKeyword.equalsIgnoreCase(splitIntoToken.nextToken())) {
+                        count++;
                     }
                 }
-                if (wordCount == 0) {
-                    System.out.println("Word does not exist");
-                } else {
-                    System.out.println("Word found");
-                    System.out.println(searchKeyword + " occurs " + wordCount + " times in the file");
-                }
+            }
+            if (count == 0) {
+                errorMessage = "Word does not exist";
+                System.out.println(errorMessage);
+                Database.databaseOperation(textFilePath, searchKeyword, searchResult = "Failure", errorMessage, count);
+            } else {
+                System.out.println("Word found");
+                System.out.println(searchKeyword + " occurs " + count + " times in the file");
+                Database.databaseOperation(textFilePath, searchKeyword, searchResult = "Success", null, count);
             }
         } catch (Exception e) {
             e.printStackTrace();
         }
     }
 }
 
+class Database {
+
+    private static String CONNECTION_URL = "jdbc:mysql://localhost:3306/elixr_labs_internship";
+    private static String USER = "root";
+    private static String PASSWORD = "abhijith.sajimon@29";
+
+    public static void databaseOperation(String textFilePath, String searchKeyword, String searchResult, String errorMessage, int count) {
+
+        try {
+            Class.forName("com.mysql.cj.jdbc.Driver");
+        } catch (ClassNotFoundException e) {
+            e.printStackTrace();
+        }
+        String insertSql = "INSERT INTO audit VALUES(?,?,now(),?,?,?)";
+        try {
+            PreparedStatement statement;
+            Connection connect = DriverManager.getConnection(CONNECTION_URL, USER, PASSWORD);
+            statement = connect.prepareStatement(insertSql);
+            statement.setString(1, textFilePath);
+            statement.setString(2, searchKeyword);
+            statement.setString(3, searchResult);
+            statement.setInt(4, count);
+            statement.setString(5, errorMessage);
+            statement.executeUpdate();
+            connect.close();

```

Comment:
What if an exception happens before this line?

Reply:
I have added a finally block to the code.